### PR TITLE
Optimize the performance of Circuit s_external mehod

### PR DESCRIPTION
**Pull request by Asachoo**
## Summary
This PR optimizes the performance of the `Circuit.s_external()` method by introducing a auxiliary matrix `[T]` that balances the **assignment efficiency** of `F-contiguous` arrays and the **computational performance** of `C-contiguous` arrays.

## Background & Motivation
`Network.s` is a 3D `ndarray` (frequency, port1 and port2). When assigning `S-parameters`, `F-contiguous` layout (column-first) is faster because assignments are grouped by ports. However, operations such as `np.matmul()`, `np.linalg.solve()`, `np.linalg.inv()`, perform faster with `C-contiguous` arrays. This PR seeks to balance these trade-offs to improve both **assignment** and **computation** efficiency.

## Key Updates
1. Added `Circuit.X_F` and `Circuit.C_F`, which use the `F-contiguous` layout, in addition to the existing `Circuit.X` and `Circuit.C`. This significantly speed up assignment for the concatenated intersection matrix `[X]` and global scattering matrix `[C]`, resolving the numerical bottleneck mentioned in `Circuit.X`.
https://github.com/scikit-rf/scikit-rf/blob/6968484b38fc35fc1d7bff06aed343687ad537ba/skrf/circuit.py#L914-L930
   
2. Introduced an auxiliary matrix `[T]` to compute `Circuit.C @ Circuit.X`. Since `[T]` uses the `F-contiguous` versions of `Circuit.C_F` and `Circuit.X_F`, directly use of `@` would cause performance issue. To prevent this, the block diagonal structure of `Circuit.X_F` is exploited, using a for-loop to reduce the size of each matrix operation and increase assignment frequency, speeding up `[T]` calculations.

3. Improved type hinting and list comprehensions for cleaner, more concise code. While these changes don't significantly impact performance, they enhance code readability.

## Results
1. Using the case from [this PR](https://github.com/scikit-rf/scikit-rf/pull/1101), I analyzed the performance of `s_external()` from `v1.0.0` to this branck and generated the following chart (501 frequency point on `OpenBLAS`/`MKL` platform):
img_v3_02ee_43aec61b-591e-4810-8d88-5b1e8d97cb9g
img_v3_02ee_c1407933-2870-4433-a543-39dfd85a303g
Benchmark code could find in this zip file.
[circuit_performance.zip](https://github.com/user-attachments/files/16883528/circuit_performance.zip)

2. Because the above case only includes **501 frequency points**, the performance improvement is not very pronounced. To provide a clearer comparison, I reran the test with my typical **2001 frequency points** (with `f_calls` = 100), and the performance comparison between `v1.3.0 (master)` and `this branch (CircuitPerformance)` is shown below:
img_v3_02ee_9bf954c0-16a0-4eba-a655-73e1f07d1dbg
img_v3_02ee_9d30c277-e0d1-4bfc-b050-9a29e5361e4g

### CODE REVIEW

**Comment by Asachoo** [comment]
By optimizing the T property with `empty_like()` initialization and leveraging `np.einsum('...ii->...i', t)[:]` for trace extraction, we've achieved a further performance improvements.
In larger circuits scale, this enhancement has slashed the `s_external()` execution time by nearly 50%. (Due to time constraints, only `MKL` environment has been tested)
Jsk0T9RCXM
fM3yOkpZaV

File: skrf/circuit.py [hunk 1]
```diff
@@ -905,20 +910,24 @@ def _Xk(self, cnx_k: list[tuple[Network, int]]) -> np.ndarray:
...
-    @cached_property
-    def X(self) -> np.ndarray:
+    def _X(self, order: MemoryLayoutT = 'C') -> np.ndarray:
         """
         Return the concatenated intersection matrix [X] of the circuit.
 
         It is composed of the individual intersection matrices [X]_k assembled
         by block diagonal.
 
+        Args:
```
Comment (jhillairet) [inline skrf/circuit.py:926]:
```suggestion
        Parameters
        ------------
```

File: skrf/circuit.py [hunk 2]
```diff
@@ -928,20 +937,110 @@ def X(self) -> np.ndarray:
         There is a numerical bottleneck in this function,
         when creating the block diagonal matrice [X] from the [X]_k matrices.
```
Comment (jhillairet) [inline skrf/circuit.py:938]: not sure this comment is still True now. Can be removed if not.

File: skrf/circuit.py [hunk 3]
```diff
@@ -928,20 +937,110 @@ def X(self) -> np.ndarray:
...
+        by block diagonal.
+
+        Returns
+        -------
+        X : :class:`numpy.ndarray`
+
+        Note
+        ----
+        There is a numerical bottleneck in this function,
+        when creating the block diagonal matrice [X] from the [X]_k matrices.
```
Comment (jhillairet) [inline skrf/circuit.py:965]: same here with the comment: still valid?

**Review by jhillairet — COMMENTED** [review]
Small comments. Sorry for the delay in reviewing.

File: skrf/circuit.py (hunk 1 above)
Comment (Asachoo) [reply skrf/circuit.py]: Fixed.

File: skrf/circuit.py (hunk 2 above)
Comment (Asachoo) [reply skrf/circuit.py]: Fixed and detailed the performance bottleneck types of 'C' order and 'F' order.

File: skrf/circuit.py (hunk 3 above)
Comment (Asachoo) [reply skrf/circuit.py:973]: The `X` property returns the `[X]` matrix in 'C' order, and thus, the comments regarding the performance bottleneck in block diagonal matrix creation remain pertinent and accurate.

**Comment by Asachoo** [comment]
> Small comments. Sorry for the delay in reviewing.

Thanks for the thorough review @jhillairet ! The comments that were not quite on point have been updated.

**Review by jhillairet — APPROVED** [review]

**Pull request merged**